### PR TITLE
Fixes unsused flags from `tsh kube exec` command

### PR DESCRIPTION
**Pull request by tigrato**
Since its introduction, `tsh kube exec` didn't have any use for the `--reason` and `--invite` flags and completely ignored what users did with it.

This PR transports those flags as query parameters to Teleport Kubernetes Service which includes them when creating the session resource in Teleport.

### CODE REVIEW

File: e
Comment (codingllama) [inline e:1]: Revert? I think this is downgrading e/.

File: lib/kube/proxy/sess.go
Comment (codingllama) [inline lib/kube/proxy/sess.go:1]: Test server-side changes?

File: e
Comment (tigrato) [reply e:1]: `git add .` is a pain :upside_down_face:

File: lib/kube/proxy/moderated_sessions_test.go [hunk 1]
```diff
@@ -429,3 +450,20 @@ func TestModeratedSessions(t *testing.T) {
 		})
 	}
 }
+
+// validateSessionTracker validates that the session tracker has the expected
+// reason and invited users.
+func validateSessionTracker(t *testing.T, testCtx *TestContext, sessionID string, reason string, invited []string) error {
```
Comment (codingllama) [inline lib/kube/proxy/moderated_sessions_test.go:456]: Drop the `t` param? I don't think we are using it. Or, alternative, drop the error response and use it?

**Review by codingllama — APPROVED** [review]

File: e
Comment (codingllama) [reply e:1]: Yep.

File: lib/kube/proxy/sess.go
Comment (codingllama) [reply lib/kube/proxy/sess.go:1]: Thanks!

**Review by tobiaszheller — APPROVED** [review]

**Pull request merged**